jupyterhub/user.py

```diff
@@ -235,8 +235,11 @@ def spawn(self, options=None):
         api_token = self.new_api_token()
         db.commit()
 
+        self.server_base_url = base_url
```

Comment:
base_url is used to set the base url inside the jupyter server, to have the same path proxy/server, with one single server is equal to `/user/{user}` but with named servers you need to change this to fit `/user/{user}/{server}.

I thought of doing this with server_base_url, or perhaps I miss something....